File: flink-demo/flinkScalaCode/src/main/scala/com/yw/flink/example/scalacases/case02_transformation/Case01_Map.scala
package com.yw.flink.example.scalacases.case02_transformation

import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment

/**
  * Flink map 测试
  */
object Case01_Map {
  def main(args: Array[String]): Unit = {
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
    import org.apache.flink.streaming.api.scala._
    val ds: DataStream[Int] = env.fromCollection(List(1, 2, 3, 4))
    ds.map(_ + 1).print()
    env.execute()
  }
}

File: flink-demo/flinkJavaCode/src/main/java/com/yw/flink/example/javacases/case02_transformation/Case01_Map.java
package com.yw.flink.example.javacases.case02_transformation;

import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

import java.util.Arrays;

/**
 * Flink map 算子测试
 */
public class Case01_Map {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        DataStreamSource<Integer> ds = env.fromCollection(Arrays.asList(1, 2, 3, 4));
        SingleOutputStreamOperator<Integer> map = ds.map(one -> one + 1);
        map.print();

        env.execute();
    }
}
